Core/Foundation/Include/Babylon/DebugTrace.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <functional>
+
+#define DEBUG_TRACE_NOOP(...)
+
+#define _STRINGIFY(x) #x
+#define _TOSTRING(x) _STRINGIFY(x)
+#define _DEBUG_TRACE(_format, ...) Babylon::DebugTrace::Trace(__FILE__ "(" _TOSTRING(__LINE__) "): " _format "\n", ##__VA_ARGS__); 
+
+#if JSRUNTIMEHOST_DEBUG_TRACE
+#   define DEBUG_TRACE _DEBUG_TRACE
+#else
+#   define DEBUG_TRACE(...) DEBUG_TRACE_NOOP()
+#endif // JSRUNTIMEHOST_DEBUG_TRACE
```

Comment:
I don't think this is enough. We probably need something that allows us to check if debug trace is enabled so that people can wrap more complex debug code in an `#ifdef`. I guess we can check `JSRUNTIMEHOST_DEBUG_TRACE`, but maybe this name isn't good?

Reply:
I've added DEBUG_RUN that takes a lambda. It works. Isn't it over kill compared to something like `assert` ? Do you have a use case for that? `JSRUNTIMEHOST_DEBUG_TRACE` might indeed not be a good name :) Naming is hard!

Reply:
using BABYLON_DEBUG_TRACE  instead. macro with lambda removed.